lib/phoenix_live_view/channel.ex

```diff
@@ -209,7 +209,23 @@ defmodule Phoenix.LiveView.Channel do
   end
 
   defp view_handle_event(%Socket{} = socket, event, val) do
-    socket.view.handle_event(event, val, socket)
+    call_live_view_handle_event!(socket, event, val)
+  end
+
+  defp call_live_view_handle_event!(%Socket{} = socket, event, val) do
+    :telemetry.span(
+      [:phoenix, :live_view, :handle_event],
+      %{socket: socket, event: event, params: val},
+      fn ->
+        case socket.view.handle_event(event, val, socket) do
+          {:noreply, %Socket{} = socket} ->
+            {{:noreply, socket}, %{socket: socket, event: event, params: val}}
+
+          other ->
+            raise_bad_callback_response!(other, socket.view, :handle_event, 3)
+        end
```

Comment:
In order to return the updated socket on the stop metric, I needed to pattern match on the response. The `ArgumentError` that's raised if the user returns an invalid response is normally raised in `handle_result/3`, but since I was already pattern matching on the response, this seemed like a good place to raise the error (and is the same pattern I followed when calling the other callbacks). In order to share the same error message code that the rest of the calls piped into `handle_result/3` use, I extracted the exception raising logic into a `raise_bad_callback_response!/4` function. That allowed me to also eliminate a function head in `handle_result/3`.

@josevalim Since this was a restructuring of the existing code, I thought I should call it out. You can see the change in more details in this commit: https://github.com/phoenixframework/phoenix_live_view/pull/983/commits/d744f5d18aca5391d81a1d265434ca8ecd049fc2. If there's a better way to do this, please let me know.

Reply:
I will review it carefully tomorrow. Note telemetry 0.4.2 is out, so we will need to use `{:telemetry, "~> 0.4.2 or ~> 0.5"}` as a dep. :)